models/amity.py

```diff
@@ -0,0 +1,428 @@
+                self.check_vacant_livingspaces()
+            self.rooms.append(new_room)
+            new_rooms.append(new_room)
+        print "You have successfully added the following rooom(s):"
+        for new_room_ in new_rooms:
+            print "Name: " + ''.join(new_room_.name) + " | Type: " \
+                + new_room_.room_type
+        print spacer
+
+    def create_room_from_db(self, args):
+        """Create new room(s) from existing database"""
+        print spacer
+        new_rooms = []
+        for room in args["<room_name>"]:
+            if room.lower() in [r.name.lower() for r in self.rooms]:
+                print "One or more rooms you tried to create already exist! " \
+                    "Please try again."
+                print spacer
+                return
+            if args["Office"]:
+                new_room = Office(room)
+                self.offices.append(new_room)
+                self.check_vacant_offices()
+            elif args["Living"]:
+                new_room = Living(room)
+                self.livingspaces.append(new_room)
+                self.check_vacant_livingspaces()
+            self.rooms.append(new_room)
+            new_rooms.append(new_room)
+        print "You have successfully added the following rooom(s):"
+        for new_room_ in new_rooms:
+            print "Name: " + ''.join(new_room_.name) + " | Type: " \
+                + new_room_.room_type
+        print spacer
+
+    def check_vacant_offices(self):
+        """Add vacant offices to lists; remove full ones from lists"""
+        for office in self.offices:
+            if len(office.occupants) < office.capacity:
+                if office not in self.vacant_offices:
+                    self.vacant_offices.append(office)
+                    self.vacant_rooms.append(office)
+            elif len(office.occupants) >= office.capacity:
+                if office in self.vacant_offices:
+                    self.vacant_offices.remove(office)
+                    self.vacant_rooms.remove(office)
+
+    def check_vacant_livingspaces(self):
+        """Add vacant living spaces to lists; remove full ones from lists"""
+        for livingspace in self.livingspaces:
+            if len(livingspace.occupants) < livingspace.capacity:
+                if livingspace not in self.vacant_livingspaces:
+                    self.vacant_livingspaces.append(livingspace)
+                    self.vacant_rooms.append(livingspace)
+            elif len(livingspace.occupants) >= livingspace.capacity:
+                if livingspace in self.vacant_livingspaces:
+                    self.vacant_livingspaces.remove(livingspace)
+                    self.vacant_rooms.remove(livingspace)
+
+    def add_person(self, args):
+        """Add new person"""
+        print spacer
+        name = args["<first_name>"] + " " + args["<last_name>"]
+        wants_space = "Yes" if args.get("<wants_space>") is "Y" else "No"
+        if args["Staff"]:
```

Comment:
**Code Quality: Nitpick**

Too much repetition here. Please rectify. Also doesn't totally follow specs. All fellows and staff are to be assigned to offices. The only optional thing is living space allocation for fellows.